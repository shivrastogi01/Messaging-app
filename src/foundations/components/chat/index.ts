export * from "./text-message/text-message";
export * from "./image-message/image-message";
export * from "./link-message/link-message";
export * from "./giphy-message/giphy-message";
export * from "./avatar/avatar";
