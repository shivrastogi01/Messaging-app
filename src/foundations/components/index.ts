export * from "./presentation";
export * from "./layout";
