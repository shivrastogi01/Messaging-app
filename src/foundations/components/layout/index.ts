export * from "./styled-box/styled-box";
export * from "./scroll-view/scroll-view";
export * from "./flex-row/flex-row";
export * from "./flex-column/flex-column";
export * from "./modal/modal";
export * from "./drawer/drawer";
export * from "./list-item/list-item";
export * from "./dropdown/dropdown";
