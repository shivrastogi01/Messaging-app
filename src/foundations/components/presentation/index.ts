export * from "./heading/heading";
export * from "./label/label";
export * from "./title/title";
export * from "./icon/icon";
export * from "./button/button";
export * from "./input/input";
export * from "./textarea/textarea";
