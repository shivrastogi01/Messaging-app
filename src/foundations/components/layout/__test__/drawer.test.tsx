import React from "react";
import renderWithTheme from "main/testing/render-with-theme";
import { waitFor } from "@testing-library/react";
import { Drawer } from "foundations/components/layout";

describe("Drawer", () => {
  test("closed matches the snapshot", () => {
    const { container } = renderWithTheme(<Drawer>Drawer content</Drawer>);
    expect(container.firstChild).toMatchSnapshot();
  });

  test("opened matches the snapshot", async () => {
    const { container } = renderWithTheme(<Drawer open>Drawer content</Drawer>);

    await waitFor(() => {
      expect(container.firstChild).toBeVisible();
      expect(container.firstChild).toHaveStyle({
        transform: "translateX(0%) translateZ(0)"
      });
    });

    expect(container.firstChild).toMatchSnapshot();
  });

  test("on right edge wide matches the snapshot", async () => {
    const { container } = renderWithTheme(
      <Drawer edge="right" wide open>
        Drawer content
      </Drawer>
    );

    await waitFor(() => {
      expect(container.firstChild).toBeVisible();
      expect(container.firstChild).toHaveStyle({
        transform: "translateX(0%) translateZ(0)"
      });
    });

    expect(container.firstChild).toMatchSnapshot();
  });
});
