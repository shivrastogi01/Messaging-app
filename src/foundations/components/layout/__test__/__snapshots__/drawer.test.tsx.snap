// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer closed matches the snapshot 1`] = `
.c0 {
  background: #FFFFFF;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100vh;
  width: 100%;
  z-index: 100;
}

@media screen and (min-width:600px) {
  .c0 {
    position: relative;
    width: 260px;
  }
}

<section
  class="c0"
  style="display: none; transform: translateX(-100%) translateZ(0);"
>
  Drawer content
</section>
`;

exports[`Drawer on right edge wide matches the snapshot 1`] = `
.c0 {
  background: #FFFFFF;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100vh;
  width: 100%;
  z-index: 100;
}

@media screen and (min-width:600px) {
  .c0 {
    position: relative;
    width: 290px;
  }
}

<section
  class="c0"
  style="display: flex; transform: translateX(0%) translateZ(0);"
>
  Drawer content
</section>
`;

exports[`Drawer opened matches the snapshot 1`] = `
.c0 {
  background: #FFFFFF;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100vh;
  width: 100%;
  z-index: 100;
}

@media screen and (min-width:600px) {
  .c0 {
    position: relative;
    width: 260px;
  }
}

<section
  class="c0"
  style="display: flex; transform: translateX(0%) translateZ(0);"
>
  Drawer content
</section>
`;
